core/simd/simd.odin

```diff
@@ -1,182 +1,2273 @@
+	b: |   0   |   2   |   2   |   2   |
+	   +-------+-------+-------+-------+
+	res:
+	   +-------+-------+-------+-------+
+	   | 0xff  | 0x00  | 0xff  | 0xff  |
+	   +-------+-------+-------+-------+
+*/
 lanes_ge :: intrinsics.simd_lanes_ge
 
+/*
+Perform a gather load into a SIMD vector.
+
+A *gather* operation is memory load operation, that loads values from an vector
+of addresses into a single value vector. This can be used to achieve the
+following results:
 
-// Gather and Scatter intrinsics
+- Accessing every N'th element of an array (strided access)
+- Access of elements according to some computed offsets (indexed access).
+- Access of elements in a different order (shuffling access).
+
+When used alongside other SIMD procedures in order to compute the offsets
+for the `ptr` and `mask` parameters.
+
+Inputs:
+- `ptr`: A vector of memory locations. Each pointer points to a single value,
+	of a SIMD vector's lane type, that will be loaded into the vector. Pointer
+	in this vector can be `nil` or any other invalid value, if the corresponding
+	value in the `mask` parameter is zero.
+- `val`: A vector of values that will be used at corresponding positions
+	of the result vector, if the corresponding memory location has been
+	masked out.
+- `mask`: A vector of booleans or unsigned integers that determines which memory
+	locations to read from. If the value at an index has the value true
+	(lowest bit set), the value at that index will be loaded into the result
+	vector from the corresponding memory location in the `ptr` vector. Otherwise
+	the value will be loaded from the `val` vector.
+
+Returns:
+
+This procedure returns a vector with all values from unmasked indices
+loaded from the pointer vector `ptr`, and all values from masked indices loaded
+from the value vector `val`.
+
+**Operation**:
+
+	for i in 0 ..< len(res) {
+		if mask[i]&1 == 1 {
+			res[i] = ptr[i]^
+		} else {
+			res[i] = val[i]
+		}
+	}
+	return res
+
+Example:
+
+Example below loads 2 lanes of values from 2 lanes of float vectors, `v1` and
+`v2`. From each of these vectors we're loading the second value, into the first
+and the third position of the result vector.
+
+Therefore the `ptrs` argument is initialized such that the first and the third
+value are the addresses of the values that we want to load into the result
+vector, and we'll fill in `nil` for the rest of them. To prevent CPU from
+dereferencing those `nil` addresses we provide the mask that only allows us
+to load valid positions of the `ptrs` array, and the array of defaults which
+will have `127` in each position as the default value.
+
+	v1 := [4] f32 {1, 2, 3, 4};
+	v2 := [4] f32 {9, 10,11,12};
+	ptrs := #simd [4]rawptr { &v1[1], nil, &v2[1], nil }
+	mask := #simd [4]bool { true, false, true, false }
+	defaults := #simd [4]f32 { 0x7f, 0x7f, 0x7f, 0x7f }
+	res := simd.gather(ptrs, defaults, mask)
+	fmt.println(res)
+
+The code would print `<2, 127, 10, 127>`. First and the third positions came
```

Comment:
You could put this output in a `Output:` block (see https://github.com/odin-lang/pkg.odin-lang.org).

Also, pretty much everywhere, you have to make sure all the pretty graphs are indented with at least a tab, so it's rendered correctly in the docs, not sure if you've done that, hard to see.